src/test/unit/types/SdrTest.cpp

```diff
@@ -29,8 +29,9 @@ TEST(SdrTest, TestConstructor) {
     // Test 0 dimensions
     EXPECT_ANY_THROW( SDR( vector<UInt>(0) ));
     // Test 0 size
-    EXPECT_NO_THROW( SDR({ 0 }) );
-    EXPECT_NO_THROW( SDR({ 3, 2, 1, 0 }) );
+    EXPECT_ANY_THROW( SDR({ 0 }) );
+    EXPECT_ANY_THROW( SDR({ 3, 2, 1, 0 }) );
```

Comment:
had to change these, I think this behavior is OK. 
Have you hacked the SDR({0}) for empty SDR in NetworkAPI, @dkeeney ?

Reply:
NetworkAPI uses SDR({0}) when creating an Array of type SDR when the size is unknown.

Reply:
> etworkAPI uses SDR({0}) 

Ok, I'll keep that. 
(Why did you not choose SDR({}) ?)

Reply:
I chose `SDR({ 0 })` over `SDR({})` because it allows the caller to specify zero cells in the SDR, and still use all of the same code-paths.  SDR({}) accepts a different # of arguments and has a different number of dimensions.

Reply:
oh, so is `SDR({0, 0, 0}) ` a valid, 3 dimensional, empty placeholder? I thought only SDR({0}) could exist.

Reply:
But SDR({1, 0, 3}) ..any combination of zero & nonzero is wrong?

Reply:
IMO mixing zeros & non-zeros should be allowed.

Reply:
> mixing zeros & non-zeros should be allowed.

what would that mean, topologically? a 3D world with 1 dimension hidden, so a 2D. We could have required 2D strictly. 
Also, our flat-size does not work, (dot-prod) is 0. Or is that what you expect?

Reply:
I'll enable all-zeros, what should be the behavior on the mix? If zeros are only for placeholders, I'd vote to require positive dimension ranges, (and only all-0 exception) Helps to avoid confusion

Reply:
As far as I am concerned, SDR({0}) is the only thing I need.  That means I don't have any dimensional information.  If I were to know that it is a 2D or 3D array, I would also probably know the actual numbers for the dimensions.

Reply:
> As far as I am concerned, SDR({0}) is the only thing I need.  

SDR({0}) is now supported and allowed by a special check